src/argilla/training/setfit.py

```diff
@@ -0,0 +1,157 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+from typing import List, Union
+
+from datasets import DatasetDict
+
+from argilla.training.utils import filter_allowed_args, get_default_args
+
+
+class ArgillaSetFitTrainer(object):
+    # @require_version("setfit", "0.6")
+    def __init__(self, dataset, device, record_class, multi_label: bool = False, seed: int = None):
+        from setfit import SetFitModel, SetFitTrainer
+
+        if seed is None:
+            seed = 42
+        self._seed = seed
+        self._record_class = record_class
+        self._multi_label = multi_label
+
+        if isinstance(dataset, DatasetDict):
+            self._train_dataset = dataset["train"]
+            self._eval_dataset = dataset["test"]
+        else:
+            self._train_dataset = dataset
+            self._eval_dataset = None
+
+        if multi_label:
+            self.multi_target_strategy = "one-vs-rest"
+            self._column_mapping = {"text": "text", "binarized_label": "label"}
+        else:
+            self.multi_target_strategy = None
+            self._column_mapping = {"text": "text", "label": "label"}
+
+        if self._multi_label:
+            self._id2label = dict(enumerate(self._train_dataset.features["label"][0].names))
+        else:
+            self._id2label = dict(enumerate(self._train_dataset.features["label"].names))
+        self._label2id = {v: k for k, v in self._id2label.items()}
+
+        self.setfit_model_kwargs = get_default_args(SetFitModel.from_pretrained)
+        self.setfit_model_kwargs["pretrained_model_name_or_path"] = "all-MiniLM-L6-v2"
+        self.setfit_model_kwargs["multi_target_strategy"] = self.multi_target_strategy
+        self.setfit_model_kwargs["device"] = device
+
+        self.setfit_trainer_kwargs = get_default_args(SetFitTrainer.__init__)
+        self.setfit_trainer_kwargs["column_mapping"] = self._column_mapping
+        self.setfit_trainer_kwargs["train_dataset"] = self._train_dataset
+        self.setfit_trainer_kwargs["eval_dataset"] = self._eval_dataset
+        self.setfit_trainer_kwargs["seed"] = self._seed
+
+    def update_config(
+        self,
+        **setfit_kwargs,
+    ):
+        """
+        It updates the `setfit_model_kwargs` and `setfit_trainer_kwargs` dictionaries with the keyword
+        arguments passed to the `update_config` function
```

Comment:
```suggestion
        Updates the `setfit_model_kwargs` and `setfit_trainer_kwargs` dictionaries with the keyword
        arguments passed to the `update_config` function.
```
Alternatively, something like "Updates config with keywords passed to `SetFitModel.from_pretrained` and `SetFitTrainer`.", as users don't know what `setfit_model_kwargs` or `setfit_trainer_kwargs` is.